### ISSUE TITLE
make fails if `./lib` is not present

### ISSUE THREAD

**Issue by alyfarahat** [uwhpsc-2016/homework2#5]
I had to  `mkdir ./lib` by hand before running make. I hope this won't be an issue when running `make test` to grade homework2.

**Comment by gadamico** [comment]
Yes.  My "make lib" is failing as well, the error message telling me that there is no such file as lib/libhomework2.so.  Would there be no problem if the file to be made were simply "libhomework2" without the "lib" dir prefix?

**Comment by jhyearsley** [comment]
I had the same problem too...

**Comment by jhyearsley** [comment]
I had the same problem too... @cswiercz ?

**Comment by jlombs** [comment]
>  Would there be no problem if the file to be made were simply "libhomework2" without the "lib" dir prefix? @gadamico 

No, the makefile sets the directory path with the LIB shortcut (which gets translated to the -L(library location) flag when compiling). So when you run the makefile, it's looking for a ./lib directory in which to stash your target library. The 'lib' in front of homework2 isn't a directory prefix, it's a pattern that is searched for specifically by the make. You can change that too, but the default flag -l(library name) tells it to look for files called lib(filename).so (for a shared object library). 

It's just coincidental that the library directory is called 'lib'. You could change the Makefile to point to 'newlib' and keep the file name the same, and you'd have the same issue until you made a ./newlib directory. 

This is my current understanding, at least...

**Comment by cswiercz** [comment]
For now just make the directory `lib` in your project. I will send out some additional inst
